modules/entity_version_history/src/Form/HistoryTabSettingsForm.php

```diff
@@ -0,0 +1,209 @@
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $entityFieldManager;
+
+  /**
+   * Constructs an HistoryTabSettingsForm object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityFieldManagerInterface $entityFieldManager
+   *   The entity field manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, EntityFieldManagerInterface $entityFieldManager) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityFieldManager = $entityFieldManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'entity_version_history_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   * @SuppressWarnings(PHPMD.NPathComplexity)
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $bundle_labels = [];
+    $entity_labels = [];
+    $field_labels = [];
+    $history_configs = [];
+    // Get entity types and bundles where the entity_version field is present.
+    $versioned_entity_types = $this->entityFieldManager->getFieldMapByFieldType('entity_version');
+
+    foreach ($versioned_entity_types as $entity_type_id => $fields) {
+      $definition = $this->entityTypeManager->getDefinition($entity_type_id);
+      if (!$definition instanceof ContentEntityTypeInterface) {
+        // We are only interested in the content entity types.
+        continue;
+      }
+      if (!$definition->hasLinkTemplate('canonical') || !$definition->hasHandlerClass('moderation')) {
+        // We are only interested in the content entity types has canonical
+        // links and are moderated.
+        continue;
+      }
+
+      // We need a list of options with labels for the form checkboxes.
+      $entity_labels[$entity_type_id] = $definition->getLabel() ?: $entity_type_id;
+
+      foreach ($fields as $field_name => $bundle_info) {
+        foreach ($bundle_info['bundles'] as $bundle_name) {
+          // We need load up the bundle and the field to prepare the checkboxes
+          // with values and labels.
+          $bundle = $this->entityTypeManager->getStorage($definition->getBundleEntityType())->load($bundle_name);
+          $bundle_labels[$entity_type_id][$bundle_name] = $bundle->label();
+          $field_config = $this->entityTypeManager->getStorage('field_config')->load("$entity_type_id.$bundle_name.$field_name");
+          $field_labels[$entity_type_id][$bundle_name][$field_name] = $field_config->label();
+
+          if ($config = HistoryTabSettings::loadByEntityTypeBundle($entity_type_id, $bundle_name)) {
```

Comment:
What about dependency injection? We just use static calls in places where we can inject services?